### PR TITLE
Change logger in PersistS3 class to Log4J

### PR DESCRIPTION
**Pull request by Pscheidl**

### CODE REVIEW

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 1]
```diff
@@ -31,10 +32,10 @@
 
 /** Persistence backend for S3 */
 public final class PersistS3 extends Persist {
-  private static final String HELP = "You can specify a credentials properties file with the -aws_credentials command line switch.";
 
   private static final String KEY_PREFIX = "s3://";
   private static final int KEY_PREFIX_LEN = KEY_PREFIX.length();
+  private static final Logger LOGGER = Logger.getLogger("water.default");
```
Comment (Pscheidl) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java:38]: @honzasterba Do we want to use this convention everywhere ? I'd put proper class in there.

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 
 /** Persistence backend for S3 */
 public final class PersistS3 extends Persist {
-  private static final String HELP = "You can specify a credentials properties file with the -aws_credentials command line switch.";
```
Comment (Pscheidl) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java:34]: Unused.

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 3]
```diff
@@ -259,8 +255,8 @@ public void importFiles(String path, String pattern, ArrayList<String> files, Ar
     }
   }
 
-  private static void ignoreAndWait(final Exception e, boolean printException) {
-    Log.ignore(e, "Hit the S3 reset problem, waiting and retrying...", printException);
+  private static void ignoreAndWait(final Exception e) {
+    LOGGER.debug("Hit the S3 reset problem, waiting and retrying...", e);
```
Comment (Pscheidl) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java:259]: I always want to print it. Former ignore was `debug` anyway. Why not print the exception all the time in debug ?

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java [hunk 4]
```diff
@@ -31,7 +31,7 @@
 
 /** Persistence backend for S3 */
 public final class PersistS3 extends Persist {
-  private static final String HELP = "You can specify a credentials properties file with the -aws_credentials command line switch.";
+  private static final Logger LOGGER = Logger.getLogger(PersistS3.class);
```
Comment (honzasterba) [inline h2o-persist-s3/src/main/java/water/persist/PersistS3.java:34]: I would rename this to LOG, same I used in xgboost

**Comment by honzasterba** [comment]
also this should go to master where we use Log4j, it will not work corectly in rel-zahrandik

File: h2o-persist-s3/src/main/java/water/persist/PersistS3.java (hunk 4 above)
Comment (Pscheidl) [reply h2o-persist-s3/src/main/java/water/persist/PersistS3.java]: Done.

**Comment by Pscheidl** [comment]
@honzasterba Not it aims at master. I'll squash it before merging into one commit.

**Pull request merged**